src/lib.rs

```diff
@@ -169,14 +171,14 @@ impl Surreal {
         Ok(())
     }
 
-    #[napi]
-    pub async fn authenticate(&self, token: String) -> Result<()> {
+    #[napi(ts_return_type="Promise<boolean>")]
+    pub async fn authenticate(&self, token: String) -> Result<Value> {
         self.db.authenticate(token).await.map_err(err_map)?;
-        Ok(())
+        Ok(Value::Bool(true))
     }
```

Comment:
The Wasm library returns a unit, like the previous implementation. By returning a boolean here, I presume when the function runs successfully it will always return `true` but never `false`, right?

Reply:
Yes, it will return `true` or it will throw an error \:)

Reply:
The issue with returning `void`, well... `undefined`, is that you cannot nicely check if the authentication succeeded. Now you can:

```js
if (await db.authenticate('...')) {
  // ...
}

// Or you can catch the error and change the response based on that
const success = await db.authenticate('...')
  .catch((error) => false);
if (success) {
```

Reply:
I see. In Rust we typically avoid using booleans in cases like that, preferring a unit instead, because a boolean gives an impression that you will need to account for a `false` response too. Which is never the case here.

Reply:
Right! We could have done that, but I don't think it's a biggie here. Typescript is not as strict as Rust (the type of `_true` is explicitly `true` here, but there are no errors or warnings about the else clause)
![image](https://github.com/surrealdb/surrealdb.node/assets/17960250/9b335a03-ac36-4dff-8413-954ef053de0a)

Reply:
It's not about compiler errors or warnings. In Rust you won't get those either when using an `if` statement. You would have to use `match` for that. The issue was the impression it gives fellow developers when looking at the return type or when looking code examples. They may feel that they need to write an else branch to be extra careful but that else branch is unnecessary and never triggered.